### DISCUSSION TITLE
CRT in offline mode issue because of "System.IdentityModel.Tokens.Jwt" dll.

### DISCUSSION THREAD

**Discussion by MichaelGMG** [microsoft/Dynamics365Commerce.Solutions#253] [Show and tell]
We have a requirement where we use the dll (System.IdentityModel.Tokens.Jwt) version of (6.23.0), but internally it is looking for (Microsoft.IdentityModel.Tokens) dll with particular version.

**Comment by ViktorVu602** [comment]
Hi @madyke,
To provide more context for this issue, customer is currently having an extension and when the POS is offline, when they login, it will show the following error on Event Viewer:
"Barcode is not valid for item. Please insert the barcode in the product configuration or update the transaction line. 
Could not find or load a specific file. (0x80131621)
File name: 'Microsoft.IdentityModel.Tokens, Version=7.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
---> System.IO.FileLoadException: Could not load file or assembly 'Microsoft.IdentityModel.Tokens, Version=7.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'"
May I ask if you can help to look into this issue? Thank you!

**Reply by madyke** [reply]
Hi @ViktorVu602 - the error message indicates that Store Commerce can't find the Microsoft.IdentityModel.Tokens file, does that file exist in their installation, or did they maybe remove it if it conflicted with the DLL they're trying to load for their extension?

Also, did they confirm that the DLL they want to use is being packaged in their installer?

**Comment by ViktorVu602** [comment]
Hi @Matt ***@***.***>,

Based on what I have seen the file does exist in their installation and they did reference the files in their code:
Here are some screenshots:
Error on event viewer:

[A screenshot of a computer  Description automatically generated]

Files in installations:

[A screenshot of a computer  Description automatically generated]
Reference on extension:
[A screenshot of a computer  Description automatically generated]
File mentioned in extension:
[A computer screen shot of a computer  Description automatically generated]
[A computer screen shot of a computer  Description automatically generated]
File version:
[A screenshot of a computer  Description automatically generated]
Hope this helps,

Best regards,
Viktor
